src/views/ClusterViz/index.js

```diff
@@ -0,0 +1,239 @@
+    }
+
+    componentWillUnmount() {
+        this.stopRequestForInitCluster();
+        this.stopRequestForClusterData();
+    }
+
+    getMaxNode = data => (
+        data.reduce((previous, current) => (
+            current.score > previous.socre ? current : previous
+        )))
+
+    clusterTableExtractor = row => row.value;
+
+    startRequestForInitCluster = () => {
+        const { activeProject } = this.props;
+        const { stopRequestForClusterData, startRequestForClusterData } = this;
+
+        const noOfClusters = 5;
+
+        const createClusterRequest = new InitProjectClusterRequest({
+            stopRequestForClusterData,
+            startRequestForClusterData,
+            setState: params => this.setState(params),
+        });
+        this.createClusterRequest = createClusterRequest.create(activeProject, noOfClusters);
+        this.createClusterRequest.start();
+    }
+
+    stopRequestForInitCluster = () => {
+        if (this.createClusterRequest) {
+            this.createClusterRequest.stop();
+        }
+    }
+
+    startRequestForClusterData = (modelId, projectId) => {
+        if (modelId) {
+            const clusterDataRequest = new ProjectClusterDataRequest({
+                setProjectClusterData: this.props.setProjectClusterData,
+                setState: params => this.setState(params),
+            });
+            this.clusterDataRequest = clusterDataRequest.create(modelId, projectId);
+            this.clusterDataRequest.start();
+        }
+    }
+
+    stopRequestForClusterData = () => {
+        if (this.clusterDataRequest) {
+            this.clusterDataRequest.stop();
+        }
+    }
+
+    headers = [
+        {
+            key: 'value',
+            label: _ts('clusterViz', 'tableValueLabel'),
+            order: 1,
+            sortable: true,
+            comparator: (a, b) => compareString(a.value, b.value),
+        },
+        {
+            key: 'score',
+            label: _ts('clusterViz', 'tableScoreLabel'),
+            order: 2,
+            sortable: true,
+            comparator: (a, b) => compareNumber(a.score, b.score),
+        },
+    ];
+
+    render() {
+        const {
+            clusterDataPending,
+            createClusterPending,
+        } = this.state;
+
+        const {
+            className,
+            activeProject,
+            projectClusterData,
+        } = this.props;
+
+        const {
+            clusterTableExtractor,
+            headers,
+        } = this;
+
+        const clusters = projectClusterData.clusterData || [];
+
+        const clusterGroup = groupList(clusters, d => d.cluster);
+
+        const nodes = clusters.map(cluster => (
+            { id: cluster.value, group: cluster.cluster }
+        ));
+
+        const maxNodes = Object.entries(clusterGroup).map(values => (
+            { id: values[0], node: this.getMaxNode(values[1]) }
+        ));
+
+        const links = [].concat(...Object.entries(clusterGroup).map((values) => {
```

Comment:
No meaning in concatenating to an empty list and using ... in this case.

why not use "Object.entries(clusterGroup).map" instead of "[].concat(...Object.entries(clusterGroup).map"